python/cuml/dask/linear_model/linear_regression.py

```diff
@@ -14,547 +13,315 @@
-        for i in range(X_df.npartitions):
-            part_number = eval(list(tt.keys())[i])[1]
-            loc_dict[part_number] = parse_host_port(str(location[i])[:-3])
-
-        # Lets divide the columns evenly, matching the order of the labels
-        part_size = ceil(X_df.shape[1] / X_df.npartitions)
-
-        # We scatter delayed operations to gather columns on the workers
-        scattered = []
-        coefs = []
-        for i in range(X_df.npartitions):
-            up_limit = min((i+1)*part_size, X_df.shape[1])
-            cols = X_df.columns.values[i*part_size:up_limit]
-            loc_cudf = X_df[cols]
-            yield wait(loc_cudf)
-            scattered.append(client.submit(preprocess_on_worker,
-                                           loc_cudf,
-                                           workers=[loc_dict[i]]))
-            yield wait(scattered)
-            coefs.append(client.submit(dev_array_on_worker,
-                                       up_limit - i*part_size,
-                                       dtype=dtype,
-                                       unique=np.random.randint(0, 1e6),
-                                       workers=[loc_dict[i]]))
-            yield wait(coefs)
-            del(loc_cudf)
-
-        # Break apart Dask.array/dataframe into chunks/parts
-        # data_parts = map(delayed, scattered)
-        data_parts = scattered
-        label_parts = y_df.to_delayed()
-        coef_parts = coefs
-
-        # Arrange parts into pairs.  This enforces co-locality
-        parts = list(map(delayed, zip(data_parts, label_parts, coef_parts)))
-        parts = client.compute(parts)  # Start computation in the background
-        yield wait(parts)
-
-        for part in parts:
-            if part.status == 'error':
-                yield part  # trigger error locally
-
-        # A dict in the form of { part_key: part }
-        key_to_part_dict = dict([(str(part.key), part) for part in parts])
-
-        who_has = yield client.who_has(parts)
-
-        worker_parts = {}
-        for key, workers in who_has.items():
-            worker = parse_host_port(first(workers))
-            if worker not in worker_parts:
-                worker_parts[worker] = []
-            worker_parts[worker].append(key_to_part_dict[key])
-
-        """
-        Create IP Handles on each worker hosting input data
-        """
-
-        # Format of input_devarrays = ([(X, y)..], dev)
-        input_devarrays = [(worker, client.submit(fit_to_device_arrays,
-                                                  part, workers=[worker]))
-                           for worker, part in worker_parts.items()]
-
-        yield wait(input_devarrays)
-
-        """
-        Gather IPC handles for each worker and call _fit() on each worker
-        containing data.
-        """
-
-        # Last worker is the only one that can have less items.
-        exec_node = loc_dict[X_df.npartitions-1]
+        Fit the model with X and y.
```

Comment:
I thouht forcing colocality was necessary to work? So the false case is supposed to be for when the user has already distributed their data correctly right? It would be valuable to explain this in the docs. Is it possible to add a check to confirm co-locality? Maybe at least look for size mismatches or worker mismatches between X and y?

Reply:
If data and label are passed to fit function like this ```fit(X["all_columns_but_labels"], X["labels"])```, then we don't need to force co-locality. If they're passed like this ```fit(X, y)```, then we need to force the co-locality.

Added couple of sentences to docstring of fit function.